examples/coap-server.c

```diff
@@ -933,6 +1417,177 @@ hnd_unknown_put(coap_context_t *ctx,
   hnd_put(ctx, r, session, request, token, query, response);
 }
 
+#if SERVER_CAN_PROXY
+
+static void
+remove_proxy_association(coap_session_t *session, int send_failure) {
+
+  size_t i;
+
+  for (i = 0; i < proxy_list_count; i++) {
+    if (proxy_list[i].incoming == session) {
+      coap_session_release(proxy_list[i].ongoing);
+      break;
+    }
+    if (proxy_list[i].ongoing == session && send_failure) {
+      if (send_failure) {
+        /* Need to send back a gateway failure */
+        coap_pdu_t *response;
+
+        response = coap_pdu_init(COAP_MESSAGE_NON,
+                                 COAP_RESPONSE_CODE(502),
+                                 coap_new_message_id(proxy_list[i].incoming),
+                            coap_session_max_pdu_size(proxy_list[i].incoming));
+        if (!response) {
+          coap_log(LOG_INFO, "PDU creation issue\n");
+          return;
```

Comment:
I am not sure if it would not be advisable to remove the association in order to avoid subsequent issues with the upstream server?

Reply:
Now doing `coap_session_release(proxy_list[i].ongoing);` if this is an upstream session being closed down